extensions/ql-vscode/src/extension.ts

```diff
@@ -1010,6 +1013,33 @@ async function activateWithInstalledDistribution(
   };
 }
 
+/**
+ * Handle changes to the external config file. This is used to restart the query server
+ * when the user changes options.
+ * See https://docs.github.com/en/code-security/codeql-cli/using-the-codeql-cli/specifying-command-options-in-a-codeql-configuration-file#using-a-codeql-configuration-file
+ */
+function watchExternalConfigFile(app: ExtensionApp, ctx: ExtensionContext) {
+  if (homedir()) {
+    const configPath = join(homedir(), ".config", "codeql", "config");
```

Comment:
(Optional) I think it would be a little bit more safe and faster to store the `homedir()` result in a variable to avoid duplicate calls.

Reply:
Sure. Done.